### PR TITLE
BYOS EAP 8 Multi VM and VMSS

### PR DESCRIPTION
**Pull request by hcherukuri**
Update the offers to support BYOS EAP 8 Multi VM and VMSS. Rename folders to generic names.

### CODE REVIEW

**Comment by backwind1233** [comment]
- we also need to update the jdk option versions and default version in [validate-byos-multivm.yaml#L18-L26](https://github.com/Azure/rhel-jboss-templates/blob/e382beaf45e20e6bb2d57af8ce466e9d0c9d6003/.github/workflows/validate-byos-multivm.yaml#L18-L26) and [validate-byos-vmss.yaml](https://github.com/Azure/rhel-jboss-templates/blob/e382beaf45e20e6bb2d57af8ce466e9d0c9d6003/.github/workflows/validate-byos-vmss.yaml#L18-L26 )

**Comment by edburns** [comment]
@backwind1233 wrote:

> Hello @edburns ,since the offer names are changed, will the azure portal link of the offers also be changed, including the aka link ?

I understand the desire for consistency, but no, we will not change that unless we absolutely must. Because the portal links correspond to the offer names, and the offer names are immutable, we need to keep those.

**Comment by hcherukuri** [comment]
@backwind1233 
Thanks for pointing out. I have updated Validation template for for VMSS and multivm.

**Comment by hcherukuri** [comment]
@backwind1233 For gracefulShutdownTimeout by default we don't set this. So think we can mark it as 0.

**Comment by backwind1233** [comment]
> @backwind1233 wrote:
> 
> > Hello @edburns ,since the offer names are changed, will the azure portal link of the offers also be changed, including the aka link ?
> 
> I understand the desire for consistency, but no, we will not change that unless we absolutely must. Because the portal links correspond to the offer names, and the offer names are immutable, we need to keep those.

Got it ,thank you for your explanation.

**Comment by backwind1233** [comment]
> @backwind1233 For gracefulShutdownTimeout by default we don't set this. So think we can mark it as 0.

Yes, I agree with you.

File: eap-rhel-byos-multivm/src/main/scripts/jbosseap-setup-master.sh [hunk 1]
```diff
@@ -49,12 +49,21 @@ function copyDomainXmlFileToShare()
...
-touch /etc/profile.d/eap_env.sh
-echo 'export EAP_HOME="/opt/rh/eap7/root/usr/share"' >> /etc/profile.d/eap_env.sh
+if [[ "${JDK_VERSION,,}" == "eap8-openjdk17" ] || [ "${JDK_VERSION,,}" == "eap8-openjdk11" ]]; then
+    export EAP_LAUNCH_CONFIG="/opt/rh/eap8/root/usr/share/wildfly/bin/domain.conf"
+    echo 'export EAP_RPM_CONF_DOMAIN="/etc/opt/rh/eap8/wildfly/eap8-domain.conf"' >> ~/.bash_profile
+    echo 'export EAP_HOME="/opt/rh/eap8/root/usr/share"' >> ~/.bash_profile
+    source ~/.bash_profile
+    touch /etc/profile.d/eap_env.sh
+    echo 'export EAP_HOME="/opt/rh/eap8/root/usr/share"' >> /etc/profile.d/eap_env.sh
+else
```
Comment (edburns) [inline eap-rhel-byos-multivm/src/main/scripts/jbosseap-setup-master.sh:59]: I would like to see a comment regarding why we don't address the case for EAP 8 with JDK 11.

**Review by edburns — APPROVED** [review]

**Pull request merged**